File: python/gtk4/procedural/application.py
import gi, sys
gi.require_version("Gtk", "4.0")
from gi.repository import Gtk

appID = "io.github.Miqueas.GTK-Examples.Python.Gtk4.Procedural.Application"

app = Gtk.Application(application_id = appID, flags = 0)
app.connect("activate", lambda _: print("Gio.Application::activate called"))
app.connect("startup", lambda _: print("Gio.Application::startup called"))

sys.exit(app.run(sys.argv))

File: cpp/gtk4/procedural/application.cpp
#include <print>
#include <gtkmm.h>

static void on_app_activate();
static void on_app_startup();

static const Glib::ustring APP_ID = "io.github.Miqueas.GTK-Examples.C.Gtk4.Procedural.ApplicationWindow";
static const Glib::ustring APP_TITLE = "Gtk::ApplicationWindow";
static Glib::RefPtr<Gtk::Application> app;

int main(int argc, char** argv) {
  app = Gtk::Application::create(APP_ID);
  app->signal_startup().connect(sigc::ptr_fun(&on_app_startup));
  app->signal_activate().connect(sigc::ptr_fun(&on_app_activate));
  return app->run(argc, argv);
}

static void on_app_activate() {
  std::println("::activate");
}

static void on_app_startup() {
  std::println("::startup");
}